### PR TITLE
CI: add workflow dispatcher for nightly GitHub workflows

### LINKED ISSUE #2092: Schedule nightly builds with a workflow dispatcher
## Why
We need to implement some refactoring and enhancements to the [Nightly backward compatibility workflow](https://github.com/input-output-hk/mithril/actions/workflows/nightly-backward-compatibility.yml) and [Nightly Docker Builds](https://github.com/input-output-hk/mithril/actions/workflows/nightly-docker-builds.yml).

## What
Implement nightly builds scheduling with a workflow dispatcher.

## How

- [x] Trigger nightly jobs with a unique scheduled workflow
- [x] Send a notification if this workflow fails:
  - [x] Investigate a way to get all core developers notified (email, GitHub notification, ...)
  - [ ] ~~e.g https://github.com/orgs/community/discussions/25351~~
- [x] Do not send error notifications when the backward compatibility workflow fails
- [x] Skip uploading of binaries in the e2e artifacts (in all workflows)

### PR DESCRIPTION
**Pull request by dlachaume**
## Content

This PR introduces a new scheduled GitHub workflow that triggers nightly tests, including Docker builds and backward compatibility testing.

Additionally, the workflow includes a step to send email notifications using SendGrid in case of Docker builds failure.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [ ] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)

Closes #2092

### CODE REVIEW

File: .github/workflows/nightly-trigger.yml
Comment (jpraynaud) [inline .github/workflows/nightly-trigger.yml:1]: Maybe you can replace `trigger` with `dispatcher` in the file name and workflow name?

File: .github/workflows/nightly-trigger.yml [hunk 1]
```diff
@@ -12,3 +12,24 @@ jobs:
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Make script executable
+        shell: bash
+        run: |
+          chmod +x ./.github/workflows/scripts/notify-nightly-failure.js
+
+      - name: Send email with SendGrid
```
Comment (jpraynaud) [inline .github/workflows/nightly-trigger.yml:29]:
```suggestion
      - name: Send failure notification by email
```

File: .github/workflows/nightly-trigger.yml [hunk 2]
```diff
@@ -12,3 +12,24 @@ jobs:
...
+        uses: peter-evans/sendgrid-action@v1
+        env:
+          SENDGRID_API_KEY: ${{ secrets.SENDGRID_API_KEY }}
+          SCRIPT_FILEPATH: ./.github/workflows/scripts/notify-nightly-failure.js
+          SENDGRID_MAIL_FROM: ${{ vars.SENDGRID_MAIL_FROM }}
```
Comment (jpraynaud) [inline .github/workflows/nightly-trigger.yml:34]:
```suggestion
          SENDGRID_MAIL_FROM: ${{ secrets.CI_NOTIFICATION_EMAIL_FROM }}
```

File: .github/workflows/nightly-trigger.yml [hunk 3]
```diff
@@ -12,3 +12,24 @@ jobs:
...
+          SENDGRID_MAIL_TO: ${{ vars.SENDGRID_MAIL_TO }}
```
Comment (jpraynaud) [inline .github/workflows/nightly-trigger.yml:35]:
```suggestion
          SENDGRID_MAIL_TO: ${{ secrets.CI_NOTIFICATION_EMAIL_TO }}
```

**Review by jpraynaud — APPROVED** [review]
LGTM :rocket:

File: .github/workflows/nightly-trigger.yml
Comment (dlachaume) [reply .github/workflows/nightly-trigger.yml:1]: You're right! 💯

File: .github/workflows/nightly-trigger.yml (hunk 1 above)
Comment (dlachaume) [reply .github/workflows/nightly-trigger.yml]: Done ✅

File: .github/workflows/nightly-trigger.yml (hunk 2 above)
Comment (dlachaume) [reply .github/workflows/nightly-trigger.yml]: Done ✅

File: .github/workflows/nightly-trigger.yml (hunk 3 above)
Comment (dlachaume) [reply .github/workflows/nightly-trigger.yml]: Done ✅

**Review by sfauvel — APPROVED** [review]
LGTM

**Review by jpraynaud — APPROVED** [review]

**Review by Alenar — APPROVED** [review]

**Pull request merged**